src/mul_acc.rs

```diff
@@ -38,4 +42,22 @@ mod tests {
         a.mul_acc(&b, &c);
         assert_eq!(a, 7.);
     }
+
+    #[derive(Debug, Copy, Clone, Default)]
+    struct Wrapped<T: Default + Copy + std::fmt::Debug>(T);
+
+    impl MulAcc<Wrapped<i32>, Wrapped<i16>> for Wrapped<i64> {
+        fn mul_acc(&mut self, a: &Wrapped<i32>, b: &Wrapped<i16>) {
+            self.0 = self.0 + a.0 as i64 * b.0 as i64;
+        }
+    }
+
+    #[test]
+    fn mul_acc_mixed_param_sizes() {
+        let mut a = Wrapped::<i64>(1i64);
+        let b = Wrapped::<i32>(2i32);
+        let c = Wrapped::<i16>(3i16);
```

Comment:
Use `i16/i32::max_value()` to show why we need the trait

Reply:
could you elaborate (use 'max_value' as the example vals, ie can't accumulate into their values, or as assertions perhaps?)
another idea might be 0x2000i16, 0x30000000i32  , mul-accumulating into the i64 (recognizable values, with products that dont fit). i might also change the test to i32+=i8*i16 for that (easier to see)

Reply:
It's enough to have something which would overflow, with max_value this would be obvious, but your choices here are good. Just make sure the sum would overflow all but the output type

Reply:
ok it now does '0x1i32 + 0x20i8 * 0x3000i16 = 0x70001i32 in the test